src/js/msp/MSPHelper.js

```diff
@@ -1452,6 +1454,42 @@ MspHelper.prototype.process_data = function(dataHandler) {
             case MSPCodes.MSP_SET_RTC:
                 console.log('Real time clock set');
                 break;
+
+            case MSPCodes.MSP_MULTIPLE_MSP:
+
+                while (data.offset < data.byteLength) {
+
+                    let command = self.mspMultipleCache.shift();
+                    let payloadSize = data.readU8();
+
+                    if (payloadSize != 0) {
+
+                        let currentDataHandler = {
+                                code         : command,
+                                dataView     : new DataView(data.buffer, data.offset, payloadSize),
+                                callbacks    : [],
+                        }
+    
+                        self.process_data(currentDataHandler);
+
+                        data.offset += payloadSize;
+                    }
+                }
+
+                // Send again MSP messages missing, the buffer in the FC was too small
+                if (self.mspMultipleCache.length > 0) {
+
+                    var partialBuffer = [];
+                    for (let i = 0; i < self.mspMultipleCache.length; i++) {
+                        partialBuffer.push8(self.mspMultipleCache[i]);
+                    }
+
+                    MSP.send_message(MSPCodes.MSP_MULTIPLE_MSP, partialBuffer, false, dataHandler.callbacks);
```

Comment:
Not sure if this won't lead to a race condition if one single command returns a result set that is too big to be returned encapsulated in `MSP_MULTIPLE_MSP`?

Reply:
I don't think one single command can do this. The difference between this and the direct command is only one byte. In my tests with an F4 I can send 20 MSP commands of different sizes an all of them returns in the same buffer.

Reply:
I can add a validation that the above if consumes some of the MSP commands and throw an error if not.

Reply:
> The difference between this and the direct command is only one byte.

Well in this case any command that returns the maximum frame length -1 will lead to an endless loop if packed into `MSP_MULTIPLE_MSP`. Since we are recursing the endless loop will eventually result in memory exhaustion and crashing. Yes, this is a small probability, but I still think we should prevent this, for example by remembering the last command we sent, and if we get to the point where we are trying to send the same again return an error.